.github/workflows/changelogs/changelogs.py

```diff
@@ -0,0 +1,639 @@
+
+        issues = []
+        # 'bnc' is the name of the tracker as defined in the trackers file
+        if 'bnc' in entry.trackers:
+            for tracker, bug_id in entry.trackers['bnc']:
+                try:
+                    bug = self.bzapi.getbug(bug_id)
+                    logging.debug(f"Bug #{bug_id} belongs to product '{bug.product}'")
+
+                    if not bug.product.startswith("SUSE Manager"):
+                        issues.append(Issue(IssueType.INVALID_PRODUCT.format(bug_id), entry.file, entry.line, \
+                                            entry.end_line, severe=False))
+                except xmlrpc.client.Fault as f:
+                    if f.faultCode == 101:
+                        # Bug not found
+                        issues.append(Issue(IssueType.BUG_NOT_FOUND.format(bug_id), entry.file, entry.line, \
+                                            entry.end_line, severe=True))
+                    elif f.faultCode == 102:
+                        # Not authorized
+                        issues.append(Issue(IssueType.BUG_NOT_AUTHORIZED.format(bug_id), entry.file, entry.line, \
+                                            entry.end_line, severe=False))
+                    else:
+                        # Any other fault
+                        issues.append(Issue(IssueType.INVALID_BUG.format(bug_id, f.faultString), entry.file, entry.line, \
+                                            entry.end_line, severe=False))
+        return issues
+
+
+    def validate_trackers(self, entries: list[type[Entry]]) -> list[type[Issue]]:
+        """Validate the trackers mentioned in a list of entries
+
+        Checks any possible typos and verifies Bugzilla trackers via the
+        Bugzilla API.
+
+        If a PR number is provided, additionally match the mentioned trackers
+        against the PR's title and comment messages.
+        """
+
+        issues = []
+
+        if self.pr_number and self.git_repo:
+            pr_validation = True
+            pr_trackers = self.get_pr_trackers(self.git_repo, self.pr_number)
+        else:
+            pr_validation = False
+
+        all_trackers = {}
+
+        for entry in entries:
+            # Check for mistyped trackers
+            # Count actual trackers in the entry
+            num_trackers = functools.reduce(lambda x, y: x + len(y), entry.trackers.values(), 0)
+            # Find all tracker-like words
+            if len(re.findall(self.regex.TRACKER_LIKE, entry.entry)) > num_trackers:
+                issues.append(Issue(IssueType.MISTYPED_TRACKER, entry.file, entry.line, entry.end_line, severe=False))
+
+            for kind, trackers in entry.trackers.items():
+                # Collect all trackers in all entries of the changelog
+                if kind not in all_trackers:
+                    all_trackers[kind] = entry.trackers[kind]
+                else:
+                    all_trackers[kind].extend(entry.trackers[kind])
```

Comment:
Another FYI: Python has syntactic sugar for this `all_trackers.setdefault(kind, []).extend(...)`.